pkg/kubernetes/interfaces.go

```diff
@@ -21,6 +22,8 @@ type Client interface {
 	ListAllPods() (*v1.PodList, error)
 	// ListNodes returns a list of Nodes registered in the api server.
 	ListNodes() (*v1.NodeList, error)
+	// ListNodesByOptions returns a list of Nodes registered in the api server.
+	ListNodesByOptions(opts metav1.ListOptions) (*v1.NodeList, error)
```

Comment:
changing this publicly exportable interface definition is dangerous FYI

Reply:
Adding methods should be fine, right?